prog/rand.go

```diff
@@ -304,6 +315,15 @@ func (r *randGen) filenameImpl(s *state) string {
 	return r.randFromMap(s.files)
 }
 
+func (r *randGen) randFilenameLength() int {
+	off := r.biasedRand(10, 5)
+	if r.bin() {
+		off = -off
+	}
+	lens := r.target.SpecialFileLenghts
+	return lens[r.Intn(len(lens))] + off
```

Comment:
Maybe add an assert that it's never below 0?

Reply:
Good point. Added a test that panics on negative result. Fixed.